### PR TITLE
Update ClusterOperator docs for changes to versions and cluster-scoped

### PR DESCRIPTION
**Pull request by jwforres**
replaces https://github.com/openshift/cluster-version-operator/pull/130

### CODE REVIEW

**Comment by jwforres** [comment]
@smarterclayton @derekwaynecarr i think this at least updates to the current state of things, it does not include anything around the Failing vs Degraded discussion

File: docs/dev/clusteroperator.md [hunk 1]
```diff
@@ -16,49 +16,101 @@ Cluster-admins make use of these resources to check the status of their clusters
...
+      version: "0.0.1-snapshot" 
 ```
 
 ## What should an operator report with ClusterOperator Custom Resource
 
+The ClusterOperator exists to communicate status about a functional area of the cluster back to both an admin
+and the higher level automation in the CVO in an opinionated and consistent way. Because of this, we document
+expectations around the outcome and have specific guarantees that apply.
+
+Of note, in the docs below we use the word `operand` to descirbe the "thing the operator manages", which might be:
```
Comment (bparees) [inline docs/dev/clusteroperator.md]: s/descirbe/describe/

File: docs/dev/clusteroperator.md [hunk 2]
```diff
@@ -16,49 +16,101 @@ Cluster-admins make use of these resources to check the status of their clusters
...
+
+1. Cause an installation to fail because a component is not able to become available for use
+2. Cause an upgrade to hang because a component is not able to successfully reach the new upgrade
+3. Prevent a user from clicking the upgrade button because components have one or more preflight criteria that are not met (e.g. nodes are at version 4.0 so the control plane can't be upgraded to 4.2 and break N-1 compat)
+4. Ensure other components are upgraded *after* your component (guarantee "happens before" in upgrades, such as kube-apiserver being updated before kube-controller-manager)
+
+There are a set of guarantees components are expected to honor in return:
+
+1. A component doesn't report the `Available` status condition the first time until they are completely rolled out (or within some reasonable percentage if the component must be installed to all nodes)
+2. A component reports `Failing` when it can't accomplish its task. This might be very broad - the API servers are down so I failed to update - or narrow - I couldn't update the last secret. In either case, Failing communicates something is wrong.
```
Comment (bparees) [inline docs/dev/clusteroperator.md]: Failing->Degraded?  (when are we doing that rename?)

Comment (jwforres) [reply docs/dev/clusteroperator.md]: i intentionally left that out of this update, we need to do a follow up when we are actually ready to roll out that change

**Comment by smarterclayton** [comment]
/lgtm

Will nitpick some stuff later.  Sorry for delay :)

**Pull request merged**